### PR TITLE
Finished various methods

### PR DESCRIPTION
**Pull request by woodepic**
Implemented methods for RB trees:
- height
- print_in_order
- print

Small fixes to CLI

I also tested all my functions via the CLI to test for operability and correctness

### CODE REVIEW

File: RBTree/src/lib.rs [hunk 1]
```diff
@@ -165,17 +167,123 @@ impl<T: Ord> RBTree<T> {
...
+        match &self.root {
+            Some(root) => {
+                let mut val = recursive_height(root);
+                val
+            },
+            None => 0
+        }
+    }
+
+    pub fn num_leaves(&self) -> u32 {
```
Comment (christianjans) [inline RBTree/src/lib.rs:196]: I think we have this function now implemented in RBTree?

Comment (christianjans) [inline RBTree/src/lib.rs:196]: I think we have this function now implemented in RBTree?

File: RBTree/src/lib.rs [hunk 2]
```diff
@@ -165,17 +167,123 @@ impl<T: Ord> RBTree<T> {
...
+                println!("{}", string_buffer.join(" "));
+            },
+            None => println!(""),
+        }
+    }
 
-    pub fn in_order() {}
+    // Inspired by:
+    // https://stackoverflow.com/questions/4965335/how-to-print-binary-tree-diagram-in-java
+    pub fn print(&self) {
```
Comment (christianjans) [inline RBTree/src/lib.rs:243]: I think we have this function now implemented in RBTree?

Comment (christianjans) [inline RBTree/src/lib.rs:243]: I think we have this function now implemented in RBTree?

**Review by christianjans — COMMENTED** [review]
Sweet thanks! Just a few comments.

**Comment by christianjans** [comment]
Sorry, I don't know why some comments appeared twice 🤦‍♂️.

**Comment by woodepic** [comment]
> Sorry, I don't know why some comments appeared twice 🤦‍♂️.

All good! I'm just looking into the conflicts now!

**Comment by christianjans** [comment]
Okay sweet, and also just ensure that the colour of the node is printed along with the node's key when we use the `print` function on the RB tree.

**Comment by woodepic** [comment]
Changes are redundant, already implemented in recent changes. No need for this PR!

**Pull request closed without merging**